### PR TITLE
Separate healthcheck task and config check task

### PR DESCRIPTION
**Pull request by shuzhang1989**
In the future, the replace_host task will only be triggered by the failure of Healthcheck task, configcheck task will only check completeness of a cluster if it's missing masters or replicas.

### CODE REVIEW

File: controller/controller-http/src/main/java/com/pinterest/rocksplicator/controller/resource/Clusters.java [hunk 1]
```diff
@@ -301,28 +302,42 @@ public Response sendLogTask(@PathParam("clusterName") String clusterName,
...
-      return Utils.buildResponse(HttpStatus.INTERNAL_SERVER_ERROR_500, ImmutableMap.of("message", message));
+      return Utils.buildResponse(HttpStatus.INTERNAL_SERVER_ERROR_500,
+          ImmutableMap.of("message", message));
     }
   }
 
+  @POST
+  @Path("/configcheck/{clusterName: [a-zA-Z0-9\\-_]+}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response configCheck(@PathParam("clusterName") String clusterName,
```
Comment (angxu) [inline controller/controller-http/src/main/java/com/pinterest/rocksplicator/controller/resource/Clusters.java]: nit: javadoc for this endpoint.

File: controller/controller-worker/src/main/java/com/pinterest/rocksplicator/controller/tasks/ConfigCheckTask.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    ClusterBean clusterBean = ZKUtil.getClusterConfig(zkClient, clusterName);
+    if (clusterBean == null) {
+      ctx.getTaskQueue().failTask(ctx.getId(), "Failed to read cluster config from zookeeper.");
+      return;
+    }
+    try {
+      for (SegmentBean segmentBean : clusterBean.getSegments()) {
+        checkSegment(segmentBean, getParameter().getNumReplicas());
+      }
+      ctx.getTaskQueue().finishTask(ctx.getId(), "Cluster " + clusterName + " has good config");
```
Comment (chengren311) [inline controller/controller-worker/src/main/java/com/pinterest/rocksplicator/controller/tasks/ConfigCheckTask.java]: String.format ?

File: controller/controller-worker/src/main/java/com/pinterest/rocksplicator/controller/tasks/HealthCheckTask.java [hunk 3]
```diff
@@ -63,14 +59,14 @@
   /**
    * Construct a new HealthCheckTask with number of replicas equals to 3
```
Comment (chengren311) [inline controller/controller-worker/src/main/java/com/pinterest/rocksplicator/controller/tasks/HealthCheckTask.java]: is the replica number still 3?

Comment (shuzhang1989) [inline controller/controller-worker/src/main/java/com/pinterest/rocksplicator/controller/tasks/HealthCheckTask.java]: good catch

**Pull request merged**